CsharpExtras/Extensions/ArrayExtension2D.cs

```diff
@@ -260,5 +263,41 @@ public static void WriteToArea<TVal>(this TVal[,] targetArray, TVal[,] arrayToWr
             }
             return result;
         }
+
+        /// <summary>
+        /// Flattens this 2D array into a 1D array using row major order.
+        public static TVal[] FlattenRowMajor<TVal>(this TVal[,] array)
+        {
+            TVal[] flattenedArray = new TVal[array.Length];
+            int numRows = array.GetLength(0);
+            int numCols = array.GetLength(1);
+            for (int row = 0; row < numRows; row++)
+            {
+                for (int col = 0; col < numCols; col++)
+                {
+                    int flatIndex = row * numCols + col;
+                    flattenedArray[flatIndex] = array[row, col];
+                }
+            }
+            return flattenedArray;
+        }
+
+        /// <summary>
+        /// Flattens this 2D array into a 1D array using column major order.
+        public static TVal[] FlattenColumnMajor<TVal>(this TVal[,] array)

```

Comment:
I think you forgot the closing `</summary>` tag for the documentation of these two methods.